src/ctelnet.cpp

```diff
@@ -1603,27 +1597,23 @@ int cTelnet::decompressBuffer( char *& in_buffer, int& length, char* out_buffer
         inflateEnd( & mZstream );
         qDebug() << "recv Z_STREAM_END, ending compression";
         this->mNeedDecompression = false;
-        this->mMCCP_version_1 = false;
-        this->mMCCP_version_2 = false;
-
-        // Reset the option state so we can re-enable compression again in the future
-        // such as in the case of a copyover -JM
-        hisOptionState[static_cast<int>(OPT_COMPRESS)] = false;
-        hisOptionState[static_cast<int>(OPT_COMPRESS2)] = false;
```

Comment:
Thing is though when those option states are false they never go back to true, and is unneccesary to keep an eye on them, if hisOptionState[foo] = false .. and then server requests it again it stays false, but you can have those, that is the only reason I just left them commented out and didn't get rid of them.  They actually don't fix copyover or anything else, perhaps they did for something a good while ago, not sure. But anyways once it's negotiated it's done for unless you plan to have more buttons for turning off and on compression by sending a WANT for it and such and hope the server listens. Other than that I don't see a reason to keep an eye on something that can be found out by .. if (OPT_COMPRESS || OPT_COMPRESS2) do this or that.. I'm not a C++ expert by any means either, I do mostly just C. But can tinker with c++ and fix things.